README.md

```diff
@@ -102,26 +102,34 @@ To set up **MacOS** for native development, complete the following steps:
 ### Docker
 
 For production hosting and deployment with **Docker**, use the `Dockerfile` and
-`docker-compose.yml` in the project root directory. To create a local
-development environment with **Docker**, complete the following steps:
+`docker-compose.yml` in the project root directory.
 
-- Install Docker Desktop
-- Run `docker compose -f .devcontainer/docker-compose.yml up -d`
-- Run `docker compose -f .devcontainer/docker-compose.yml exec app bin/setup`
-- Finally, run `docker compose -f .devcontainer/docker-compose.yml exec app bin/dev`
+For local development, install and launch [Docker Desktop], and run:
 
-If you are using an IDE with [support for the Development Container specification](https://containers.dev/supporting), it will run the above `docker compose` commands automatically. For **Visual Studio Code** this requires the [Dev Container extension](https://containers.dev/supporting#dev-containers).
+```shell
+docker compose -f .devcontainer/compose.yaml up -d
+docker compose -f .devcontainer/compose.yaml exec app bin/setup
+docker compose -f .devcontainer/compose.yaml exec app bin/dev
+```
+
+Within IDEs that support the [Development Containers] specification, the above
+`docker compose` commands should run automatically. For **Visual Studio Code**
+this requires installing the [Dev Container extension].
 
 ### GitHub Codespaces
 
-To get you coding in just a few minutes, GitHub Codespaces provides a web-based version of Visual Studio Code and a cloud-hosted development environment fully configured with the software needed for this project..
+[GitHub Codespaces] provides a web-based version of VS Code and a cloud hosted
+development environment configured with the software needed for this project.
+
+[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)][codespace]
 
-- Click this button to create a new codespace:<br>
-  [![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://github.com/codespaces/new?hide_repo_select=true&ref=main&repo=52281283&devcontainer_path=.devcontainer%2Fcodespaces%2Fdevcontainer.json)
-- Wait for the environment to build. This will take a few minutes.
+- Click the button to create a new codespace, and confirm the options.
+- Wait for the environment to build (takes a few minutes).
 - When the editor is ready, run `bin/dev` in the terminal.
-- After a few seconds, a popup will appear with a button labeled _Open in Browser_. This will open Mastodon.
-- On the _Ports_ tab, right click on the “stream” row and select _Port visibility_ → _Public_.
+- Wait for a prompt to appear with a button labeled _Open in Browser_. This will
+  open Mastodon.
+- On the _Ports_ tab, right click on the “stream” row and select _Port
```

Comment:
Is this something we can automate somehow?

Reply:
I don't see a documented way https://docs.github.com/en/codespaces/developing-in-a-codespace/forwarding-ports-in-your-codespace#sharing-a-port - seems to just indicate doing manually via that menu.